### ISSUE TITLE
OAuth Token is missing response when trying to use any function

### ISSUE THREAD

**Issue by ankoure** [Freguglia/rTwitchAPI#4]
When I try to work through the example I receive this response. Unclear where the Oauth token is supposed to be added. 

Error in get_streams(first = 15, language = "en") : 
  OAuth token is missing

**Comment by Freguglia** [comment]
Hi @ankoure, thanks for letting me know. It happens that the Twitch API now requires an OAuth token for the endpoints that are called within the package functions, but our authentication engine doesn't setup an OAuth token as they were not required previously.

I'll be working on it as soon as I get a chance (probably tomorrow), but it should be a very simple change to implement. For a reference: https://dev.twitch.tv/docs/authentication. Right now, the package is missing the steps 2 and 3, as they were not required for the endpoints we use, but now they are, so 2 and 3 must be implemented with the `twitch_auth()` function.

**Comment by Freguglia** [comment]
Hi @ankoure, thanks for letting me know. Did you setup your credentials using the `twitch_auth()` function as specified in the Authenticantion section of the README file?

You can setup an OAuth token by calling the `twitch_auth()` function after you setup `TWITCH_CLIENT_ID` and `TWITCH_CLIENT_SECRET` as environment variables.

**Comment by ankoure** [comment]
I added the TWITCH_CLIENT_ID and TWITCH_CLIENT_SECRET to the r environ file, then restarted R. I used the twitch_auth() function and it ran without an error but it did not prompt anything additional. Is twitch_auth() supposed to prompt something or should I be using the auth token in the function?

**Comment by Freguglia** [comment]
`twitch_auth()` handles the authentication process automatically, so you never really have to deal with the tokens manually. What it does in the backend is: 

  * a `POST` request to `https://id.twitch.tv/oauth2/token` is sent using the credentials (`TWITCH_CLIENT_ID` and `TWITCH_CLIENT_SECRET`) found in your `.Renviron` file.
  * an access token is then created by the Twitch API and it is sent to you in the response of the request, as described in the [Client credentials flow](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth/#oauth-client-credentials-flow)
  * the received access token is read and passed to `httr`'s `set_config()` so that it will be automatically passed as a header in future requests made by `httr`.
  * now any request made by the package's functions (e.g. `get_streams()`) is made with the token as a header.

In summary, [`twitch_auth()`](https://github.com/Freguglia/rTwitchAPI/blob/master/R/authentication.R) obtains the token for you and automatically adds it to every future request made by `rTwitchAPI` functions. If it ran without any error, then the other functions should work normally.

I think I'll add some kind of feedback in `twitch_auth()` so that is becomes clearer that something happened in the background and you're now able to use the other functions.
